src/l500/ac-trigger.cpp

```diff
@@ -1211,34 +1238,9 @@ namespace ivcam2 {
     {
         auto hwm = _hwm.lock();
         if( ! hwm )
-        {
-            AC_LOG( ERROR, "Hardware monitor is inaccessible; cannot read temperature" );
-            return 0.;
-        }
-        // The temperature may depend on streaming?
-        std::vector<byte> res;
+            throw std::runtime_error( "HW monitor is inaccessible - stopping algo" );
```

Comment:
Why change the behavior?

Reply:
we said that I will lock the hwm to protect the case that the device will no longer exist.

Reply:
I still don't understand why you changed the behavior of `read_temperature()`